transport/http/integration_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/encoding/json"
+	"go.uber.org/yarpc/internal/clientconfig"
+	pkgerrors "go.uber.org/yarpc/pkg/errors"
+)
+
+func TestBothResponseError(t *testing.T) {
+	tests := []struct {
+		inboundBothResponseError  bool
+		outboundBothResponseError bool
+	}{
+		{
+			inboundBothResponseError:  false,
+			outboundBothResponseError: false,
+		},
+		{
+			inboundBothResponseError:  false,
+			outboundBothResponseError: true,
+		},
+		{
+			inboundBothResponseError:  true,
+			outboundBothResponseError: false,
+		},
+		{
+			inboundBothResponseError:  true,
+			outboundBothResponseError: true,
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
No worries either way: I think I would accept a nested for loop instead of generating the cross-product above.

Reply:
Ack - I think this is fine